b2sdk/sync/path.py

```diff
@@ -0,0 +1,74 @@
+######################################################################
+#
+# File: b2sdk/sync/path.py
+#
+# Copyright 2021 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+from abc import ABC, abstractmethod
+from typing import List
+
+from ..file_version import FileVersionInfo
+
+
+class AbstractSyncPath(ABC):
+    """
+    Represent a path in a source or destination folder - be it B2 or local
+    """
+
+    def __init__(self, relative_path: str, mod_time: int, size: int):
+        self.relative_path = relative_path
+        self.mod_time = mod_time
+        self.size = size
+
+    @abstractmethod
+    def is_visible(self):
+        """Is the path visible/not deleted on it's storage"""
+
+    def __repr__(self):
+        return '%s(%s, %s, %s)' % (
+            self.__class__.__name__, repr(self.relative_path), repr(self.mod_time), repr(self.size)
+        )
+
+
+class LocalSyncPath(AbstractSyncPath):
+    __slots__ = ['relative_path', 'mod_time', 'size']
+
+    def is_visible(self):
+        return True
+
+
+class B2SyncPath(AbstractSyncPath):
+    __slots__ = ['relative_path', 'selected_version', 'all_versions']
+
+    def __init__(
+        self, relative_path: str, selected_version: FileVersionInfo, all_versions: List[FileVersionInfo]
```

Comment:
well, B2SyncPath does have different attributes then LocalSyncPath, but they share common interface (attributes/properties). Not really sure how to handle it otherwise

Reply:
I can live with that if @ppolewicz is ok